File: src/dmcpworkflow/exec_dmcp.py
import argparse
import numpy as np
from src.pycv.dmcp import dmcp, dm_to_world

if __name__ == "__main__":
    parser = argparse.ArgumentParser(description='execute dmcp on ushichka')
    parser.add_argument("--imK")
    parser.add_argument("--imP")
    parser.add_argument("--dmK")
    parser.add_argument("--dmP")
    parser.add_argument("--dmIm")
    parser.add_argument("--cps")
    parser.add_argument("--out")


    args = parser.parse_args()

    # read images and set origin to bottom left
    imK = np.loadtxt(args.imK, delimiter=",")
    imP = np.loadtxt(args.imP, delimiter=",")

    dmK = np.loadtxt(args.dmK, delimiter=",")
    dmP = np.loadtxt(args.dmP, delimiter=",")
    dmIm = np.loadtxt(args.dmIm, delimiter=",")[-1:0:-1,:]

    cps = np.loadtxt(args.cps, delimiter=",")

    world_points = dm_to_world(dmIm, dmK,dmP,cps[:,2:])

    A = dmcp(imK, imP, cps[:,:2], world_points)

    np.savetxt(args.out, A, delimiter=',') 
    print(f"transformation saved to {args.out}")

File: src/dmcpworkflow/_exec_dmcp.jl
include("src/dmcp_alg.jl")
using ArgParse

function parse_commandline()

    s = ArgParseSettings()
    
    @add_arg_table s begin
        "--imK"
            arg_type = String
            required = true
        "--imP"
            arg_type = String
            required = true
        
        "--dmK"
            arg_type = String
            required = true
        "--dmP"
            arg_type = String
            required = true
        "--dmIm"
            arg_type = String
            required = true

        "--cps"
            arg_type = String
            required = true
        
        "--out", "-o"
            arg_type = String
            default = "transform.csv"
        
        "--outPdlt"
            arg_type = String
            default = "Pdlt.csv"

    end

    return parse_args(ARGS, s)
end

parsed_args = parse_commandline()

using DelimitedFiles

## reading input
#println("reading input files")
imK = readdlm(parsed_args["imK"], ',')
imP = readdlm(parsed_args["imP"], ',')

dmK = readdlm(parsed_args["dmK"], ',')
dmP = readdlm(parsed_args["dmP"], ',')
dmIm = readdlm(parsed_args["dmIm"], ',')[end:-1:1, 1:end]

cps = readdlm(parsed_args["cps"], ',')

# convert types
dmIm = convert(Matrix{Float32}, dmIm)
cps = convert(Matrix{Float64}, cps)

#println("executing algorithm")
Pdlt, transform = exec_dmcp(imK, imP, dmIm, dmK, dmP, cps)

writedlm(parsed_args["outPdlt"], Pdlt, ',')
writedlm(parsed_args["out"], transform, ',') 
println("transformation saved to $(parsed_args["out"])")